test/widgets/compose_box_test.dart

```diff
@@ -1239,4 +1245,41 @@ void main() {
         maxHeight: verticalPadding + 170 * 1.5, maxVisibleLines: 6);
     });
   });
+
+  group('ComposeBoxState new-event-queue transition', () {
+    testWidgets('content input not cleared when store changes', (tester) async {
+      // Regression test for: https://github.com/zulip/zulip-flutter/issues/1470
+
+      TypingNotifier.debugEnable = false;
+      addTearDown(TypingNotifier.debugReset);
+
+      final channel = eg.stream();
+      await prepareComposeBox(tester,
+        narrow: eg.topicNarrow(channel.streamId, 'topic'), streams: [channel]);
+
+      await enterContent(tester, 'some content');
+      checkContentInputValue(tester, 'some content');
+
+      store.updateMachine!
+        ..debugPauseLoop()
+        ..poll()
+        ..debugPrepareLoopError(
+            eg.apiExceptionBadEventQueueId(queueId: store.queueId))
+        ..debugAdvanceLoop();
+      await tester.pump();
+
+      // new store replaces the old one
+      final newStore = testBinding.globalStore.perAccountSync(eg.selfAccount.id)!;
+      check(store).not((it) => it.identicalTo(newStore));
+
+      // new store has the same basic boring data so the compose box
+      // has its content input instead of no-posting-permission error banner
+      // TODO instead, add data by preparing new store's initial snapshot
+      await newStore.addStream(channel);
+      await newStore.addUser(eg.selfUser);
```

Comment:
Maybe the TODO just calls for moving setup code into the initial snapshot in `prepareComposeBox`?

```diff
     await testBinding.globalStore.add(selfAccount, eg.initialSnapshot(
+      realmUsers: [selfUser, ...otherUsers],
+      streams: streams,
       zulipFeatureLevel: zulipFeatureLevel,
       realmMandatoryTopics: mandatoryTopics,
     ));
 
     store = await testBinding.globalStore.perAccount(selfAccount.id);
-
-    await store.addUsers([selfUser, ...otherUsers]);
-    await store.addStreams(streams);
```

The initial call to `testBinding.globalStore.add` sets the initial snapshot that will also be used when reloading the store.

Reply:
> The initial call to `testBinding.globalStore.add` sets the initial snapshot that will also be used when reloading the store.

Ohh, I didn't realize; thanks for pointing this out.